emily_cron/app/models/deposit.py

```diff
@@ -48,16 +48,21 @@ def from_json(cls, j: dict[str, Any]) -> Self:
     @property
     def lock_time(self) -> int:
         """Extracts lock time from reclaim script."""
-        lock_time = script.parse(self.reclaim_script)[0]
-        if lock_time.startswith("OP_"):
-            return int(lock_time[len("OP_") :])
-        return int.from_bytes(bytes.fromhex(lock_time), byteorder="little")
+        script = Script.parse(self.reclaim_script)
+        op_code_maybe = script.view().split()[0]
+        if op_code_maybe.startswith("OP_"):
+            return int(op_code_maybe[len("OP_") :])
+        return int.from_bytes(script.commands[0], byteorder="little")
 
     @property
     def max_fee(self) -> int:
         """Extracts the max fee from deposit script."""
-        data = script.parse(self.deposit_script)[0]
-        return int.from_bytes(bytes.fromhex(data)[:8])
+        script = Script.parse(self.deposit_script)
+        if script.redeemscript:
+            max_fee_bytes = script.redeemscript
+        else:
+            max_fee_bytes = bytes.fromhex(script.view().split()[0])
```

Comment:
qq: how is `redeemscript` computed, and why we don't get it always (and we need the `else`)?